state-chain/chains/src/sol/transaction_builder.rs

```diff
@@ -113,18 +114,23 @@ impl SolanaTransactionBuilder {
 				match param.asset {
 					SolAsset::Sol => {
 						compute_limit += COMPUTE_UNITS_PER_FETCH_NATIVE;
+						let fetch_pda_and_bump = derive_fetch_account(
+							param.deposit_fetch_id.address,
+							sol_api_environment.vault_program,
+						)
+						.map_err(SolanaTransactionBuildingError::FailedToDeriveAddress)?;
+
+						// Add extra compute units for each bump derivation required on-chain
+						compute_limit += (AccountBump::MAX - fetch_pda_and_bump.bump) as u32 *
+							COMPUTE_UNITS_PER_BUMP_DERIVATION;
```

Comment:
Why is it `(AccountBump::MAX - fetch_pda_and_bump.bump)`, intuitively I'd have thought it to be just `fetch_pda_and_bump.bump * COMPUTE_UNITS_PER_BUMP_DERIVATION`

Reply:
because bumps are started in reverse. So from the max value towards zero.

Reply:
I think this will be easier to understand if we move the implementation into `PdaAndBump` impl as mentioned above (maybe with a comment to explain).

Reply:
The `PdaAndBump` is part of the Solana primitives while this logic is more of a compute unit estimation (gas estimation) when crafting a transaction to be broadcasted. It might not be as straight forward to understand but putting it as part of an address primitive is conceptually wrong. Therefore, I've added it as a function to the TransactionBuilder.